# PhD-thesis-statistical-analyses

Bioinformatics pipelines for microbial amplicon sequence data (demultiplexed, paired-end reads) generated from gut microbial communities of Aotearoa New Zealand kākāpō and takahē.

Link to thesis (update once published).
***

## README.md table of contents

1.1 Bacteria_16S_pipeline_chapter_three

1.2 Fungi_ITS_pipeline_chapter_four

1.3 Bacteria_16S_pipeline_chapter_five

1.4 GWAS_microbiome_pipeline_chapter_five

1.5 Bacteria_16S_pipeline_chapter_six

***

## General notes

Data chapters in the thesis 'The microbiome in threatened species conservation: implications for the conservation of kākāpō and takahē':

- Chapter Three: Influence of management practice on the microbiota of a critically endangered species: A longitudinal study of kākāpō chick faeces and associated nest litter
- Chapter Four: The mycobiota of faeces from kākāpō chicks and associated nest litter
- Chapter Five: Capturing species-wide diversity of the gut microbiota and its relationship with genomic variation in the critically endangered kākāpō
- Chapter Six: Gut microbiota of the threatened takahē: biogeographic patterns and conservation implications

All chapters utilised amplicon sequence variants (ASVs) produced by the R package [DADA2](https://benjjneb.github.io/dada2/). 

***

##Publications from this PhD thesis

**Chapter Two**: [The microbiome in threatened species conservation](https://www.sciencedirect.com/science/article/pii/S0006320718311145)

**Chapter Three**: 

**Chapter Four**:

**Chapter Five**:

**Chapter Six**: [Gut microbiota of the threatened takahē: biogeographic patterns and conservation implications](https://animalmicrobiome.biomedcentral.com/articles/10.1186/s42523-021-00158-5)
